jobs/pulse3d/test-pulse3d-worker/src/main.py

```diff
@@ -0,0 +1,360 @@
+            # if a new name has been given in the upload form, then replace here, else use original name
+            analysis_filename = (
+                f"{name_override}.zip"
+                if (name_override := metadata.get("name_override"))
+                else upload_filename
+            )
+        except Exception:
+            logger.exception("Fetching upload details failed")
+            raise
+
+        with tempfile.TemporaryDirectory(dir="/tmp") as tmpdir:
+            zipped_recording_filename = f"{os.path.splitext(upload_filename)[0]}.zip"
+            zipped_recording_key = f"{prefix}/time_force_data/{PULSE3D_VERSION}/{zipped_recording_filename}"
+            zipped_recording_path = os.path.join(tmpdir, zipped_recording_filename)
+
+            stim_waveforms_path = os.path.join(tmpdir, "stim_waveforms.parquet")
+            tissue_waveforms_path = os.path.join(tmpdir, "tissue_waveforms.parquet")
+            metadata_path = os.path.join(tmpdir, "metadata.json")
+
+            features_parquet_key = f"{prefix}/{job_id}/peaks_valleys.parquet"
+            features_filepath = os.path.join(tmpdir, "peaks_valleys.parquet")
+
+            try:
+                key = f"{prefix}/{upload_filename}"
+                recording_path = f"{tmpdir}/{analysis_filename}"
+                s3_client.download_file(PULSE3D_UPLOADS_BUCKET, key, recording_path)
+                logger.info(f"Downloaded recording file to {recording_path}")
+            except Exception:
+                logger.exception("Failed to download recording zip file")
+                raise
+
+            try:
+                # attempt to download peaks and valleys from s3, will only be the case for interactive analysis jobs
+                s3_client.download_file(PULSE3D_UPLOADS_BUCKET, features_parquet_key, features_filepath)
+                interactive_analysis = True
+                logger.info(f"Downloaded peaks and valleys to {features_filepath}")
+            except Exception:  # TODO catch only boto3 errors here
+                logger.info("No existing peaks and valleys found for recording")
+
+            try:
+                s3_client.download_file(PULSE3D_UPLOADS_BUCKET, zipped_recording_key, zipped_recording_path)
+                logger.info(f"Downloaded existing preanalyed data to {zipped_recording_path}")
+
+                with ZipFile(zipped_recording_path) as z:
+                    z.extractall(tmpdir)
+
+                tissue_waveforms = pl.read_parquet(tissue_waveforms_path)
+                stim_waveforms = (
+                    pl.read_parquet(stim_waveforms_path) if os.path.exists(stim_waveforms_path) else None
+                )
+
+                metadata_dict = json.load(open(metadata_path))
+                is_beta_2 = metadata_dict["file_format_version"] >= VersionInfo.parse("1.0.0")
+                existing_metadata = (
+                    MantarrayBeta2Metadata(**metadata_dict)
```

Comment:
Do we have support for nautilus files yet?

Reply:
yes, true

Reply:
@tannermpeterson does the new way make sense?

Reply:
Yeah I guess since we don't have a Nautilus metadata class we'll need to use BaseMetadata for now. Eventually we could make a function that takes the instrument type in the metadata dict or something and spits out the correct metadata class